### ISSUE TITLE
[TorchInductor] `remove_noop_ops` misses a full-range slice whose negative start normalizes to zero

### ISSUE THREAD

**Issue by nikokurumi** [pytorch/pytorch#196358]
### 🐛 Describe the bug

`remove_noop_ops` removes an `aten.slice` that selects an entire dimension when `start=0`, but misses the equivalent negative-index form `start=-dim_size`.

In `torch/_inductor/fx_passes/post_grad.py`, `slice_noop` recognizes the beginning of a full-range slice only when `start` is literally zero. A negative start that normalizes to zero is therefore not eliminated.

## Minimal reproducer

```python
import torch


class Model(torch.nn.Module):
    def forward(self, x):
        y = torch.ops.aten.slice.Tensor(x, 0, -4, 4, 1)
        return y + 1


def main():
    x = torch.randn(4, 8, device="cuda")

    eager = Model()(x)

    torch._dynamo.reset()
    compiled = torch.compile(Model(), fullgraph=True)
    actual = compiled(x)

    torch.testing.assert_close(actual, eager)
    print("PASS")


if __name__ == "__main__":
    main()
```

The first dimension has size 4 and is equivalent to:

```python
torch.ops.aten.slice.Tensor(x, 0, 0, 4, 1)
```

## Observed behavior

Tracing with:

```bash
TORCH_TRACE=trace_path python test.py
tlparse trace_path -o tlparse_path
```

shows that the negative-index full-range slice is present before the post-grad pipeline:

```python
slice_1 = torch.ops.aten.slice.Tensor(arg0_1, 0, -4, 4)
```

and remains present in the generated post-grad FX graph.

The relevant `slice_noop` logic checks the raw start bound against zero, effectively requiring:

```python
start == 0
```

Therefore `start=-4` is rejected even though, for a dimension of size 4, it normalizes to zero and the slice covers the full dimension.

### Versions

Reproduced on PyTorch main commit: `e289091`

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo
